web/src/message_events.js

```diff
@@ -35,86 +34,6 @@ import * as unread_ops from "./unread_ops";
 import * as unread_ui from "./unread_ui";
 import * as util from "./util";
 
-function maybe_add_narrowed_messages(messages, msg_list, callback, attempt = 1) {
```

Comment:
moved this function to a temporary utils file named `message_events_util.js` to avoid import cycle error when importing this function directly in `echo.js`.

<details><summary>Dependency cycle error</summary>
<img src="https://github.com/user-attachments/assets/71f17089-8888-45c7-934f-d76877a48060">
</img>
</details> 

Added a TODO comment which we should do post 9.0:
> // TODO: Move this function to 'message_util.ts' once #30702 is merged.

Reply:
It might actually be better for this to be in a separate module, let me consider.

Reply:
Yeah I think `message_util` would be a fine home for this, so this comment is good.